src/DocumentManager.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_DOCUMENTMANAGER_H
+#define HTTPLIB_SRC_DOCUMENTMANAGER_H
+
+#include <string>
+#include <vector>
+
+#include "DocumentSession.h"
+
+class DocumentSession;
+
+/** @brief Store document modify revision */
+typedef struct{
+    /** @brief Who made this revision. */
+    std::string userId;
+    /** @brief Relative insert position from document start and end. */
+    int startPosition;
+    int endPosition;
+    /** @brief insertData and deleteLength will only exist one at the same time. */
+    std::string insertData;
+    int deleteLength;
+} Revision;
+
+typedef struct{
+    /** @brief Document id. */
+    int id;
+    /** @brief Modify history list. */
+    std::vector<Revision> history;
+} Document;
+
+/** @brief Document Manager */
+class DocumentManager {
+private:
+    /** @brief Map recording id and DocumentSession. */
+    std::map<id, DocumentSession*> docMap_;
```

Comment:
all lowercase,like doc_map_ or doc_map
reference https://google.github.io/styleguide/cppguide.html#Variable_Names

Reply:
Yes. This is a reasonable review.

Reply:
Also note, `id` doesn't seemed to be a type. Does this build?

Reply:
ok, `id` misstyped